src/Core/src/AbstractApi.php

```diff
@@ -226,9 +226,15 @@ private function getSigner(?string $region)
         /** @var string $region */
         $region = $region ?? ($this->configuration->isDefault('region') ? null : $this->configuration->get('region'));
         if (!isset($this->signers[$region])) {
-            $metadata = $this->getEndpointMetadata($region);
             $factories = $this->getSignerFactories();
             $factory = null;
+            if (!$this->configuration->isDefault('endpoint') && !$this->configuration->isDefault('region')) {
+                $metadata = $this->getEndpointMetadata(Configuration::DEFAULT_REGION);
+                $metadata['signRegion'] = $region;
+            } else {
+                $metadata = $this->getEndpointMetadata($region);
+            }
```

Comment:
You could invert this if-statement to improve readability


```php

```


```suggestion
            if ($this->configuration->isDefault('endpoint') || $this->configuration->isDefault('region')) {
                $metadata = $this->getEndpointMetadata($region);
            } else {
                // Allow non-aws region with custom endpoint
                $metadata = $this->getEndpointMetadata(Configuration::DEFAULT_REGION);
                $metadata['signRegion'] = $region;
            }
```